docs/source/API/core/policies/TeamHandleConcept.rst

```diff
@@ -0,0 +1,227 @@
+``TeamHandleConcept``
+=====================
+
+.. role::cpp(code)
+    :language: cpp
+
+.. role:: cppkokkos(code)
+    :language: cppkokkos
+
+Header File: ``<Kokkos_Core.hpp>``
+
+TeamHandleConcept defines the concept for the ``member_type`` of ``TeamPolicy`` and ``TeamTask``.
+The actual type is defined through the policies, but meets the following API requirements.
+Note that the specific classes are only part of the public API as provided by ``TeamPolicy`` and 
+``TeamTask`` and only their parts as defined by the ``TeamHandleConcept``. 
+The actual name of the class, as well as potential template parameters, existing
+constructors, member functions going beyond the concept, etc., are NOT part of the public Kokkos API
+and are thus subject to change. 
+
+Synopsis 
+--------
+
+.. code-block:: cpp
+
+    class TeamHandleConcept {
+        public:
+        // Nested aliases
+        using execution_space      = /*see details in the description below*/;
+        using scratch_memory_space = /*see details in the description below*/;
+
+        // Constructor, Destructor, Assignment
+        TeamHandleConcept( TeamHandleConcept && ) = default ;
+        TeamHandleConcept( TeamHandleConcept const & ) = default ;
+        ~TeamHandleConcept() = default ;
+        TeamHandleConcept & operator = ( TeamHandleConcept && ) = default ;
+        TeamHandleConcept & operator = ( TeamHandleConcept const & ) = default ;
+
```

Comment:
In comparing the "Before" and "After," is the key difference showing _only_ the public API in the .rst version?  Assuming this is the case, I like it : )

That being said, I do think the section headings (e.g., Index Queries, etc.) should be used.

Reply:
We discussed with @fnrizzi and yes I am now using section headings.

The end result will be like the original .md file but with the specifications for functions starting with `KOKKOS_INLINE_FUNCTION`.

I haven't updated the screenshots because you have to validate the request #256 first.

=> The section titles are back, no worries.

Reply:
To answer your question, yes, the purpose is to show the visual difference in order to already validate if the new visual is correct.

Then the code can be inspected by the Reviewers. As the visual is the most important for the future users and for the Reviewers who validate or not the changes, it is much more convenient for you to already show the final result.